core/templates/pages/exploration-editor-page/exploration-editor-page.component.ts

```diff
@@ -499,6 +506,48 @@ angular.module('oppia').component('explorationEditorPage', {
             }
           )
         );
+        ctrl.directiveSubscriptions.add(
+          ConnectionService.onInternetStateChange.subscribe(
+            currentState => {
+              ctrl.hasNetworkConnection = currentState.hasNetworkConnection;
+              ctrl.hasInternetAccess = currentState.hasInternetAccess;
+              if (ctrl.hasNetworkConnection && ctrl.hasInternetAccess) {
+                if (ctrl.status === 'OFFLINE') {
+                  if (
+                    !ChangeListService.isExplorationLockedForEditing() &&
+                    !ctrl.countWarnings()) {
+                    $(
+                      '.oppia-editor-publish-button'
+                    ).removeAttr('disabled');
```

Comment:
Please avoid using jQuery DOM manipulation here and below. We should have existing ways to enable/disable the publish buttons, please use that.

Reply:
Removed it.